File: AMD AI IIT DELHI/AMD AI IIT DELHI/eval.py
import json
import sys
import argparse

def evaluate(q_file, a_file):
    try:
        with open(q_file, "r") as f:
            questions = json.load(f)
        with open(a_file, "r") as f:
            answers = json.load(f)
        
        N = len(questions)
        correct = 0
        details = []
        
        for i, (q, a) in enumerate(zip(questions, answers)):
            is_correct = q['answer'] == a['answer']
            if is_correct:
                correct += 1
            details.append({
                "id": i + 1,
                "topic": q['topic'],
                "q_ans": q['answer'],
                "a_ans": a['answer'],
                "status": "PASS" if is_correct else "FAIL"
            })
        
        print("\n" + "="*40)
        print(" AUTOMATED PROJECT RUN REPORT ")
        print("="*40)
        print(f"Total Questions Processed: {N}")
        print(f"Correct Multi-Choice:    {correct}")
        print(f"Accuracy Rate:           {(correct/N)*100:.2f}%")
        print("="*40)
        
        # Display breakdown by topic
        topics = {}
        for d in details:
            t = d['topic']
            if t not in topics: topics[t] = {"total": 0, "correct": 0}
            topics[t]["total"] += 1
            if d['status'] == "PASS": topics[t]["correct"] += 1
        
        print("\nTOPIC BREAKDOWN:")
        for t, stats in topics.items():
            rate = (stats['correct']/stats['total'])*100
            print(f"- {t:40} | {rate:6.2f}% ({stats['correct']}/{stats['total']})")
        print("="*40 + "\n")
        
    except Exception as e:
        print(f"Evaluation failed: {e}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--q_file", default="outputs/questions.json")
    parser.add_argument("--a_file", default="outputs/answers.json")
    args = parser.parse_args()
    
    evaluate(args.q_file, args.a_file)

File: AMD AI IIT DELHI/AMD AI IIT DELHI/eval.js
const fs = require('fs');

function evaluate(qFile, aFile) {
    try {
        const questions = JSON.parse(fs.readFileSync(qFile, 'utf8'));
        const answers = JSON.parse(fs.readFileSync(aFile, 'utf8'));

        const N = questions.length;
        let correct = 0;
        const details = [];

        for (let i = 0; i < N; i++) {
            const q = questions[i];
            const a = answers[i];
            const isCorrect = q.answer === a.answer;
            if (isCorrect) correct++;
            details.push({
                id: i + 1,
                topic: q.topic,
                qAns: q.answer,
                aAns: a.answer,
                status: isCorrect ? "PASS" : "FAIL"
            });
        }

        console.log("\n" + "=".repeat(40));
        console.log(" AUTOMATED PROJECT RUN REPORT ");
        console.log("=".repeat(40));
        console.log(`Total Questions Processed: ${N}`);
        console.log(`Correct Multi-Choice:    ${correct}`);
        console.log(`Accuracy Rate:           ${((correct / N) * 100).toFixed(2)}%`);
        console.log("=".repeat(40));

        const topics = {};
        details.forEach(d => {
            const t = d.topic;
            if (!topics[t]) topics[t] = { total: 0, correct: 0 };
            topics[t].total++;
            if (d.status === "PASS") topics[t].correct++;
        });

        console.log("\nTOPIC BREAKDOWN:");
        for (const [t, stats] of Object.entries(topics)) {
            const rate = (stats.correct / stats.total) * 100;
            console.log(`- ${t.padEnd(40)} | ${rate.toFixed(2).padStart(6)}% (${stats.correct}/${stats.total})`);
        }
        console.log("=".repeat(40) + "\n");

    } catch (e) {
        console.error(`Evaluation failed: ${e.message}`);
    }
}

const qFile = process.argv[2] || "outputs/full_pipeline_questions.json";
const aFile = process.argv[3] || "outputs/full_pipeline_answers.json";

evaluate(qFile, aFile);
